### PR TITLE
A Tale of Blobs and Man

### PR DESCRIPTION
**Pull request by Mechoid**
Human mobs with AIs can attack, AI mobs consider blobs to be enemies unless they're Blob faction.
- rscadd: "New blobs! Prepare to die."
- rscadd: "Ravenous Macrophage (Medium): Highly resistant to burn, produces non-opaque toxic smoke when prey is near. Deals bioacid damage checking bio."
- rscadd: "Roiling Mold (Medium): Weak to brute and burn. Factories can fire 'mortars' up to 10 tiles away which spray acid on impact. Deals brute checking melee."
- rscadd: "Ectoplasmic Horror (Medium): Critically weak to brute and burn. Can 'dodge' attacks, and factories cause exhaustion if you remain within their sight. Deals halloss damage checking anomaly resistance, method of murder is the slow-produced spores."
- tweak: "Volatile Alluvium tweaked to not be hilariously awful to fight. Smoke is no longer opaque, and water does more damage. Spores given projectile dispersion."
- tweak: "AI mobs now check blobs as possible enemies."
- tweak: "Human mobs with AI now instantiate a HuD."
- rscadd: "Human mobs controlled by AIs can 'click' on things, specifically to attack, grab, disarm, etc."
- rscadd: "You can now punch blobs. It doesn't do much usually (Unless you have Gauntlets or knuckledusters), but you can."

### CODE REVIEW

**Comment by zekesulastin** [comment]
We MIGHT want to look carefully at making artillery blobs "medium" difficulty unless there's a way to target said artillery through the blob itself.

**Comment by Mechoid** [comment]
They travel slow enough that it's dodgeable, and having something like a voidsuit will protect you from the acid splash. I can up the difficulty of it if needed.

**Comment by Nalarac** [comment]
Your PR is on the example map you goof
image

**Comment by Mechoid** [comment]
Intentional, still working on this. ..I should put the WIP on.

File: code/modules/blob2/overmind/types.dm [hunk 1]
```diff
@@ -98,6 +132,14 @@
 /datum/blob_type/grey_goo/on_emp(obj/structure/blob/B, severity)
 	B.adjust_integrity(-(20 / severity))
 
+/datum/blob_type/grey_goo/on_chunk_tick(obj/item/weapon/blobcore_chunk/B)
```
Comment (Atermonera) [inline code/modules/blob2/overmind/types.dm:135]: Each supbtype should probably go into separate files

File: code/modules/blob2/blobs/core.dm [hunk 2]
```diff
@@ -78,6 +78,15 @@ var/list/blob_cores = list()
 /obj/structure/blob/core/volatile_alluvium
 	desired_blob_type = /datum/blob_type/volatile_alluvium
 
+/obj/structure/blob/core/ravenous_macrophage
+	desired_blob_type = /datum/blob_type/ravenous_macrophage
+
+/obj/structure/blob/core/roiling_mold
+	desired_blob_type = /datum/blob_type/roiling_mold
+
+/obj/structure/blob/core/ectoplasmic_horror
```
[the remark this answers is not in the document]
Comment (Neerti) [reply code/modules/blob2/blobs/core.dm:87]: The meta blob events spawn the 'choose a random blob between X and Y difficulty' blob cores so it should be able to use them automatically.

File: code/modules/blob2/core_chunk.dm [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+			should_tick = TRUE
+		else if(blob_type.chunk_active_type == BLOB_CHUNK_TOGGLE)
+			should_tick = FALSE
+
+		active_ability_cooldown = blob_type.chunk_active_ability_cooldown
+		passive_ability_cooldown = blob_type.chunk_passive_ability_cooldown
+
+		blob_type.chunk_setup(src)
+
+		START_PROCESSING(SSobj, src)
```
[the remark this answers is not in the document]
Comment (Mechoid) [reply code/modules/blob2/core_chunk.dm:55]: 'should_tick' only defines if it should be actively trying to get the parent blob type to do something. IE, if the toggle is enabled, or disabled.

**Pull request merged**